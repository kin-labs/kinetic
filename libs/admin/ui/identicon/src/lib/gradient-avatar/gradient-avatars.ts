import { parseSize } from './helper'
import { generateGradient, svgToDataURL } from './image'

export interface GenerateSvgOptions {
  name: string
  text?: string
  width?: string
  height?: string
}

export function generateSvg(options: GenerateSvgOptions) {
  const { name, text, height, width } = {
    height: '512',
    text: '',
    width: '512',
    ...options,
  }
  return generateGradient(name, text, parseSize(width), parseSize(height))
}

export function generateSvgDataUrl(options: GenerateSvgOptions) {
  return svgToDataURL(generateSvg(options))
}
