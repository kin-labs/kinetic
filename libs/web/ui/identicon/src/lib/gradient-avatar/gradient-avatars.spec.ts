import { generateSvg } from './gradient-avatars'

describe('gradientAvatars', () => {
  it('should generate an SVG', () => {
    expect(
      generateSvg({
        name: 'random-seed',
        text: 'text',
        width: '512',
        height: '512',
      }),
    ).toMatchSnapshot()
  })
})
