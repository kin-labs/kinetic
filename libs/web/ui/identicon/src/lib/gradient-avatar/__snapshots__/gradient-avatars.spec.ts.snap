// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gradientAvatars should generate an SVG 1`] = `
"<?xml version=\\"1.0\\" standalone=\\"no\\"?>
<!DOCTYPE svg PUBLIC \\"-//W3C//DTD SVG 1.1//EN\\" \\"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\\">
<svg width=\\"512\\" height=\\"512\\" viewBox=\\"0 0 512 512\\" version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <g>
    <defs>
      <linearGradient id=\\"avatar\\" x1=\\"0\\" y1=\\"0\\" x2=\\"1\\" y2=\\"1\\">
        <stop offset=\\"0%\\" stop-color=\\"#FAFF13\\"/>
        <stop offset=\\"100%\\" stop-color=\\"#49DFA4\\"/>
      </linearGradient>
    </defs>
    <rect fill=\\"url(#avatar)\\" x=\\"0\\" y=\\"0\\" width=\\"512\\" height=\\"512\\"/>
    <text x=\\"50%\\" y=\\"50%\\" alignment-baseline=\\"central\\" dominant-baseline=\\"central\\" text-anchor=\\"middle\\" fill=\\"#fff\\" font-family=\\"sans-serif\\" font-size=\\"115.2\\">text</text>
  </g>
</svg>
"
`;
